### PR TITLE
Fix issues found by lgtm

### PR DESCRIPTION
**Pull request by daurnimator**

### CODE REVIEW

**Comment by daurnimator** [comment]
@sjvs the https://lgtm.com/rules/7890079/ warning seems to be misfiring?

The code on master was wrong:
```js
api_check(L, n|0 === n && 1 <= n && n <= f.nupvalues, "invalid upvalue index");
```

So I fixed it to:
```js
api_check(L, (n|0) === n && 1 <= n && n <= f.nupvalues, "invalid upvalue index");
```

However this was still coming up as a warning in lgtm. I also tried this:
```js
api_check(L, ((n|0) === n) && (1 <= n) && (n <= f.nupvalues), "invalid upvalue index");
```

But the alert remains.

File: src/lparser.js [hunk 1]
```diff
@@ -640,8 +640,7 @@ const open_func = function(ls, fs, bl) {
     fs.nactvar = 0;
     fs.firstlocal = ls.dyd.actvar.n;
     fs.bl = null;
-    let f = new Proto(ls.L);
-    f = fs.f;
+    let f = fs.f;
```
Comment (giann) [inline src/lparser.js]: We don't need a new object here ?

Comment (daurnimator) [reply src/lparser.js]: It was immediately thrown away. `fs.f` already contains a `Proto`

**Comment by sjvs** [comment]
@daurnimator: as far as I can tell you haven't actually merged the code that resolves [the LGTM alert](https://lgtm.com/projects/g/fengari-lua/fengari/snapshot/3841e9332581c7b251aaaed8eb03db1a15d6a169/files/src/lapi.js#x3cc7de9dcc1a80f5:1) into `master` yet. LGTM only analyses master (and any pull requests into it), so after this PR has been merged the alert should disappear.

Note that in some cases it may take up to 24 hours for LGTM to notice your change due to the fact that we can't constantly poll all 80,000 repositories.

**Comment by sjvs** [comment]
Ah, this just got merged. LGTM alerts should update soon!

**Comment by sjvs** [comment]
I'll have a look — stay tuned!

**Comment by sjvs** [comment]
The alert is definitely gone — I'm looking into why that wasn't reported during LGTM's automatic code review. Thanks for pointing this out @daurnimator!

**Pull request merged**